### DISCUSSION TITLE
Progress Made on Postgresql.

### DISCUSSION THREAD

**Discussion by iampopal** [simolus3/drift#3106] [General]
We have been using Drift for a long time and are very happy using this package. thank you @simolus3 for this amazing package.

I am thinking about a new project to start and use PostgreSQL, I have some questions in mind to ask:
1) SQLite is by default Installed in Windows and mobile (iOS, Android), is there a way to by default install PostgreSQL in the Windows Operating System as the user installs our flutter application?
2) Are all PostgreSQL Types like ( varchar, datetimez, jsonb ) and other special types supported.

**Comment by simolus3** [comment]
PostgreSQL is designed as a network DBMS following a client-server architecture. I haven't heard about people shipping a Postgres copy to run a local server on each client before. It's possible, but most would probably just use sqlite3 for those use cases.

> SQLite is by default Installed in Windows and mobile (iOS, Android), is there a way to by default install PostgreSQL in the Windows Operating System as the user installs our flutter application?

Unlike SQLite, Postgres is not a library that can be part of your application. It needs to run in its own process which you'd have to start in your application. You probably also need your own configuration for Postgres so that it doesn't conflict with other apps.

For client-side databases as part of your application, I strongly suggest using sqlite3 over Postgres. Even if you need some features that aren't available in sqlite3, getting halfway there with user-defined functions or other customizations is likely still a better experience.

> Are all PostgreSQL Types like ( varchar, datetimez, jsonb ) and other special types supported.

We support the ones you've mentioned, but not all of them. Note that you can also get most of them to work on sqlite3 with type converters though.

**Reply by iampopal** [reply]
Thank you for your awesome replay.
